src/main/java/xyz/nifeather/morph/network/multiInstance/slave/NetworkDataHolder.java

```diff
@@ -50,106 +48,152 @@ public NetworkDataHolder(SlaveInstance bindingSlave)
     }
 
     @Override
-    public List<DisguiseMeta> getAvailableDisguisesFor(Player player)
+    public CompletableFuture<PlayerMeta> loadAsync(UUID uuid)
     {
-        var playerMeta = getPlayerMeta(player);
+        var future = new CompletableFuture<PlayerMeta>();
+
+        bindingSlave.requestData(uuid).thenAccept(u ->
+        {
+            var data = nullablePlayerMeta(u);
 
-        return playerMeta.getUnlockedDisguises();
+            if (data != null)
+                future.complete(data);
+            else
+                future.completeExceptionally(new NullDependencyException("The future of the requested data has been finished, but we can't find a matching data in NetworkDataHolder!"));
+        });
+
+        return future;
     }
 
+    /**
+     * Get or load data for the given UUID
+     *
+     * @param uuid
+     */
     @Override
-    public CompletableFuture<PlayerMeta> loadPlayerDataAsync(UUID uuid)
+    public CompletableFuture<PlayerMeta> getOrLoad(UUID uuid)
     {
+        var existing = nullablePlayerMeta(uuid);
+        if (existing != null)
+            return CompletableFuture.completedFuture(existing);
+
         var future = new CompletableFuture<PlayerMeta>();
 
-        bindingSlave.requestData(uuid).thenAccept(u ->
+        bindingSlave.requestData(uuid);
+        bindingSlave.getOrCreatePlayerFuture(uuid).thenAccept(u ->
         {
-            var data = nullablePlayerMeta(Bukkit.getOfflinePlayer(u));
-
+            var data = this.nullablePlayerMeta(u);
             if (data != null)
                 future.complete(data);
             else
-                future.completeExceptionally(new NullDependencyException("The future of the requested data has been finished, but we can't find a matching data in NetworkDataHolder!"));
+                future.completeExceptionally(new RuntimeException("Request for %s has been finished, but we can't find it in an instance of NetworkDataHolder".formatted(uuid)));
         });
```

Comment:
Same for here as above.

Reply:
`getOrLoad` now calls `loadAsync` when the NetworkDataHolder doesn't have a matching data stored